app/mailers/school_mailer.rb

```diff
@@ -66,33 +69,27 @@ def nomination_confirmation_email(user:, school:, start_url:, step_by_step_url:)
         subject: "Sign in to manage induction",
         step_by_step: step_by_step_url,
       },
-    )
+    ).tag(:sit_nominated).associate_with(school, sit_profile)
   end
 
-  def school_partnership_notification_email(
-    recipient:,
-    lead_provider_name:,
-    delivery_partner_name:,
-    school_name:,
-    nominate_url:,
-    challenge_url:,
-    challenge_deadline:
-  )
+  # This email is sent to the school which has been reported by the lead provider and which hasn't yet nominated SIT
+  # It also contains request to nominate SIT.
+  def school_partnership_notification_email(recipient:, partnership:, challenge_url:, nominate_url:)
     template_mail(
       SCHOOL_PARTNERSHIP_NOTIFICATION_EMAIL_TEMPLATE,
       to: recipient,
       rails_mailer: mailer_name,
       rails_mail_template: action_name,
       personalisation: {
-        lead_provider_name: lead_provider_name,
-        delivery_partner_name: delivery_partner_name,
-        school_name: school_name,
+        lead_provider_name: partnership.lead_provider.name,
+        delivery_partner_name: partnership.delivery_partner.name,
+        school_name: partnership.school.name,
```

Comment:
How much coupling are we comfortable introducing here? This is probably fine, but I don't think I'd want any more

Reply:
Agree - I'd much rather prefer having associations and tagging available outside of the mailer, so more like:

```
SchoolMailer.some_email().associate_with(school).deliver_later
```
While possible, that requires quite a lot of monkey patching